ddprof-lib/src/main/cpp/javaApi.cpp

```diff
@@ -113,22 +113,26 @@ Java_com_datadoghq_profiler_JavaProfiler_execute0(JNIEnv *env, jobject unused,
 
 extern "C" DLLEXPORT jstring JNICALL
 Java_com_datadoghq_profiler_JavaProfiler_getStatus0(JNIEnv* env,
-                                                    jobject unused) {
+                                                    jclass unused) {
   char msg[2048];
   int ret = Profiler::instance()->status((char*)msg, sizeof(msg) - 1);
   return env->NewStringUTF(msg);
 }
 
 extern "C" DLLEXPORT jlong JNICALL
 Java_com_datadoghq_profiler_JavaProfiler_getSamples(JNIEnv *env,
-                                                    jobject unused) {
+                                                    jclass unused) {
   return (jlong)Profiler::instance()->total_samples();
 }
 
 // some duplication between add and remove, though we want to avoid having an extra branch in the hot path
+
+// JavaCritical is faster JNI, but more restrictive - parameters and return value have to be
+// primitives or arrays of primitive types.
+// We direct corresponding JNI calls to JavaCritical to make sure the parameters/return value
```

Comment:
Seems like a word is missing. Maybe this?

```suggestion
// We add direct corresponding JNI calls to JavaCritical to make sure the parameters/return value
```